packages/tools/kadena-cli/src/services/config/config.service.ts

```diff
@@ -37,8 +45,53 @@ export interface IConfigService {
 }
 
 export class ConfigService implements IConfigService {
-  // eslint-disable-next-line @typescript-eslint/naming-convention, @typescript-eslint/parameter-properties
-  public constructor(private services: Services) {}
+  public directory: string | null = null;
+  public constructor(
+    // eslint-disable-next-line @typescript-eslint/naming-convention, @typescript-eslint/parameter-properties
+    private services: Services,
+    directory?: string,
+  ) {
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    this.setDirectory(directory);
+  }
+
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  private async setDirectory(directory?: string): Promise<void> {
```

Comment:
in the config.ts file the logic was specifically written to be sync, and while it being in a service gives somewhat more flexibility, this is still dangerous since class constructors can not be async.

doing for exmaple:
```ts
process.env.ENV_KADENA_DIR = "/whatever"; // (valid directory)
const services = new Services();
console.log(services.config.getDirectory());
```
will probably be `null`, since the async `setDirectory()` did not finish yet.

There's two options to solve it:
1. what I usually do when I want async in a constructor is introduce a `init()` method, not called within the constructor, so you have to do 
```ts
const services = new Services();
await services.init()
```
that's tricky in this case, since services is by default a singleton you can import and use directly, a top-level await could be used in ESM, but I'm not so experienced in it's downsides.

2. make `setDirectory` not async, it was already that way, so it's not hard to do. The downside is that it's less clean, we have the `directoryExists` service method available, seems odd not to use it.

---
I personally prefer option 2.